config/embedded-languages.nix

```diff
@@ -0,0 +1,112 @@
+# Embedded Language Support Configuration
+# Enables LSP and syntax highlighting for embedded Lua and Bash code within Nix files
+
+{ pkgs, ... }:
+{
+  config.vim = {
+    # Configure treesitter injections for embedded languages in Nix files
+    treesitter = {
+      grammars = pkgs.vimPlugins.nvim-treesitter.allGrammars;
+    };
+
+    # Custom Lua configuration for treesitter injections and otter-nvim
+    luaConfigRC.embedded-languages = ''
+      -- Configure treesitter injections for Nix files
+      
+      -- Enhanced injection for Lua code in setup strings
+      vim.treesitter.query.set("nix", "injections", [[
+        ; Lua injection for setup = ''' ... ''' (primary pattern)
+        (binding
+          attrpath: (attrpath (identifier) @_key (#eq? @_key "setup"))
+          expression: (indented_string_expression 
+            (string_fragment) @injection.content)
+          (#set! injection.language "lua"))
+          
+        ; Lua injection for nested setup in extraPlugins
+        (apply_expression
+          function: (select_expression
+            attrpath: (attrpath (identifier) @_setup (#eq? @_setup "setup")))
+          argument: (indented_string_expression 
+            (string_fragment) @injection.content)
+          (#set! injection.language "lua"))
+      ]])
+      
+      -- Enhanced injection for Bash code in writeShellScriptBin
+      vim.treesitter.query.set("nix", "injections", [[
+        ; Bash injection for writeShellScriptBin "name" ''' ... '''
+        (apply_expression
+          function: (select_expression
+            expression: (variable) @_pkgs (#eq? @_pkgs "pkgs")
+            attrpath: (attrpath (identifier) @_func (#eq? @_func "writeShellScriptBin")))
+          argument: [
+            (string_expression)
+            (indented_string_expression 
+              (string_fragment) @injection.content)
+          ]
+          (#set! injection.language "bash"))
```

Comment:
Can we modify these ingjections to be more generaric so that they automatically attempt to detect the embedded language for code blocks and don't rely on the setup or writeShellScriptBin examples?